optimism/src/keccak/lookups.rs

```diff
@@ -182,4 +171,12 @@ impl<Fp: Field> Lookups for KeccakEnv<Fp> {
             }
         }
     }
+
+    fn lookup_rc16(&mut self, rw: LookupMode, flag: Self::Variable, value: Self::Variable) {
+        self.add_lookup(Lookup {
+            numerator: Signed::new(rw, Some(flag)),
```

Comment:
I personally find this less clear than the alternative. Perhaps it makes more sense to expand the `Lookup` structure to become:
```rust
Lookup<F> {
    mode: LookupMode,
    /// The number of times that this lookup value should be added to / subtracted from the lookup accumulator.
    magnitude_contribution: F,
    table_id: LookupTable,
    value: Vec<F>,
}
```

Reply:
Ha! Good point 🤔

Reply:
Addressed here! https://github.com/o1-labs/proof-systems/pull/1653